task1/task1.go

```diff
@@ -0,0 +1,44 @@
+package task1
+
+import (
+	"strconv"
+	"unicode/utf8"
+)
+
+func contains(val string) bool {
+	for _, a := range []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"} {
+		if val == a {
+			return true
+		}
+	}
+	return false
+}
+
+func getIntStr(s string) string {
+	if utf8.RuneCountInString(s) == 0 {
+		return "0"
+	}
+	res := ""
+	for _, r := range s {
+		if contains(string(r)) {
+			res = res + string(r)
+			continue
+		}
+		if !contains(string(r)) && utf8.RuneCountInString(res) == 0 {
+			continue
+		}
+		if !contains(string(r)) && utf8.RuneCountInString(res) > 0 {
+			return res
+		}
+	}
+	return res
+}
+
+func StrToInt(s string) (int, error) {
+	line := getIntStr(s)
```

Comment:
I think we should pass `s` string into the Atoi function without any additional preparations. Atoi will take care about invalid values.

Reply:
Agree with Ivan, it's better to see an error in case of type instead of remove some unexpected chars

Reply:
I just wanted to do an abracadabra spell
as Sergey Shtripling wrote [here](https://epamgolangcourses2019.slack.com/archives/CQFG3J954/p1574945221010700), well, at least a mini spell
but ok, i fixed it